src/ddprof_worker.cc

```diff
@@ -575,9 +654,19 @@ DDRes ddprof_worker_process_event(const perf_event_header *hdr, int watcher_pos,
     case PERF_RECORD_SAMPLE:
       if (wpid->pid) {
         uint64_t mask = ctx->watchers[watcher_pos].sample_type;
+        bool is_allocation =
+            (ctx->watchers[watcher_pos].type == kDDPROF_TYPE_CUSTOM &&
+             ctx->watchers[watcher_pos].config == kDDPROF_COUNT_ALLOCATIONS);
+        if (is_allocation) // temp hack
+          mask |= PERF_SAMPLE_ADDR;
         perf_event_sample *sample = hdr2samp(hdr, mask);
         if (sample) {
```

Comment:
Here I need @sanchda 's PR to improve this code. The aim is to avoid aggregation when the watcher does not need it. David introduces such a mechanism.